connectors/mongo/conn.go

```diff
@@ -232,13 +232,15 @@ func (c *conn) GeneratePlan(ctx context.Context, r *connect.Request[adiomv1.Gene
 				return nil
 			}
 			numSamples := (count / c.settings.TargetDocCountPerPartition) * int64(c.settings.SampleFactor)
-			if numSamples*20 >= count {
-				slog.Warn("Too many samples requested, adjusting...", "original", numSamples, "new", count/20)
-				numSamples = count / 20
+			if numSamples*21 >= count {
+				return fmt.Errorf("too many samples requested (%v of %v)- adjust sample factor or target documents per partition", numSamples, count)
 			}
-			res, err := col.Aggregate(ctx, mongo.Pipeline{{{"$sample", bson.D{{"size", numSamples}}}}, {{"$sort", bson.D{{"_id", 1}}}}})
+			if numSamples > 1000000 {
+				slog.Warn("More than 1000000 samples requested", "samples", numSamples)
```

Comment:
Maybe elaborate on why it's bad (can cause the sampling query to run long or fail) and what to do (adjust the sampling factor or increase task size